requests/urllib3 installation broken in Dockerfile.db

I tried running the deployment process because eoAPI sounds AWESOME but I hit a snag on the database deployment. I got it working but it's worth a PR to fix it.

#### Problem description

A recent upgrade to `urllib3` seems to have broken the version of `requests` that gets installed in Dockerfile.db so the `BootstrappedDb` deployment fails.

I can't seem to get a full traceback from AWS CloudWatch but here is the error message:
```sh
[ERROR] Runtime.ImportModuleError: Unable to import module 'handler': cannot import name 'DEFAULT_CIPHERS' from 'urllib3.util.ssl_' (/var/task/urllib3/util/ssl_.py)
```

I was able to fix the problem by explicitly installing `urllib3<2` in Dockerfile.db:
```diff
diff --git a/infrastructure/aws/dockerfiles/Dockerfile.db b/infrastructure/aws/dockerfiles/Dockerfile.db
index 52df1b4..17c4d23 100644
--- a/infrastructure/aws/dockerfiles/Dockerfile.db
+++ b/infrastructure/aws/dockerfiles/Dockerfile.db
@@ -10,7 +10,7 @@ WORKDIR /tmp
 RUN python -m pip install pip -U
 
 RUN echo "Using PGSTAC Version ${PGSTAC_VERSION}"
-RUN python -m pip install requests psycopg["binary,pool"] pypgstac==${PGSTAC_VERSION} -t /asset
+RUN python -m pip install urllib<2 requests psycopg["binary,pool"] pypgstac==${PGSTAC_VERSION} -t /asset
 
 COPY infrastructure/aws/handlers/db_handler.py /asset/handler.py
```

Here is a relevant issue in `requests`: https://github.com/psf/requests/issues/6432
`requests` v2.30.0 [should be working with `urllib3>2`](https://github.com/psf/requests/releases/tag/v2.30.0) but maybe we are pinned to something less than v2.30.0 by another package. I haven't run the docker network locally yet but will try to dig in to find out the actual best solution.


#### Environment Information

Ubuntu 22.04, Python 3.10.6

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -40,7 +40,7 @@ jobs:
 
       - name: Install python dependencies
         run: |
-          python -m pip install pytest pytest-asyncio httpx pypgstac==0.7.1 psycopg[pool] brotlipy requests boto3 pytest-pgsql psycopg2
+          python -m pip install pytest pytest-asyncio httpx pypgstac==0.7.1 psycopg[pool] brotlipy boto3 pytest-pgsql psycopg2
 
       - name: Test CDK DB Bootstrap
         working-directory: ./infrastructure/aws

--- a/infrastructure/aws/dockerfiles/Dockerfile.db
+++ b/infrastructure/aws/dockerfiles/Dockerfile.db
@@ -1,4 +1,4 @@
-ARG PYTHON_VERSION=3.9
+ARG PYTHON_VERSION=3.10
 
 FROM public.ecr.aws/lambda/python:${PYTHON_VERSION}
 
@@ -10,7 +10,7 @@ WORKDIR /tmp
 RUN python -m pip install pip -U
 
 RUN echo "Using PGSTAC Version ${PGSTAC_VERSION}"
-RUN python -m pip install requests psycopg["binary,pool"] pypgstac==${PGSTAC_VERSION} -t /asset
+RUN python -m pip install httpx psycopg["binary,pool"] pypgstac==${PGSTAC_VERSION} -t /asset
 
 COPY infrastructure/aws/handlers/db_handler.py /asset/handler.py
 

--- a/infrastructure/aws/handlers/db_handler.py
+++ b/infrastructure/aws/handlers/db_handler.py
@@ -1,14 +1,17 @@
 """Bootstrap Postgres db."""
 
 import json
+import logging
 
 import boto3
+import httpx
 import psycopg
-import requests
 from psycopg import sql
 from psycopg.conninfo import make_conninfo
 from pypgstac.db import PgstacDB
 from pypgstac.migrate import Migrate
+
+logger = logging.getLogger("eoapi-bootstrap")
 
 
 def send(
@@ -56,10 +59,12 @@ def send(
     headers = {"content-type": "", "content-length": str(len(json_responseBody))}
 
     try:
-        response = requests.put(responseUrl, data=json_responseBody, headers=headers)
-        print("Status code: " + response.reason)
+        response = httpx.put(responseUrl, data=json_responseBody, headers=headers)
+        print(f"Status code: {response.status_code}")
+        logger.debug(f"OK - Status code: {response.status_code}")
     except Exception as e:
-        print("send(..) failed executing requests.put(..): " + str(e))
+        print("send(..) failed executing httpx.put(..): " + str(e))
+        logger.debug(f"NOK - failed executing PUT requests:  {e}")
 
 
 def get_secret(secret_name):

--- a/infrastructure/aws/tests/test_bootstrap.py
+++ b/infrastructure/aws/tests/test_bootstrap.py
@@ -1,14 +1,15 @@
 """Test bootstrap handler."""
 
 import json
+import logging
 from unittest.mock import patch
 
 from handlers import db_handler
 
 
-@patch("handlers.db_handler.requests")
+@patch("handlers.db_handler.httpx")
 @patch("handlers.db_handler.boto3")
-def test_bootstrap(boto3, requests, database_url):
+def test_bootstrap(boto3, httpx, database_url, caplog):
     """Test Bootstrap."""
     boto3.client.return_value.get_secret_value.side_effect = [
         # connection_params
@@ -38,11 +39,12 @@ def test_bootstrap(boto3, requests, database_url):
     ]
 
     class put:
-        """Fake requests.put response."""
+        """Fake httpx.put response."""
 
+        status_code: int = 200
         reason: str = "All Good"
 
-    requests.put.return_value = put()
+    httpx.put.return_value = put()
 
     event = {
         "StackId": "eoapi-test",
@@ -66,4 +68,7 @@ def test_bootstrap(boto3, requests, database_url):
 
     context = ctx()
 
-    db_handler.handler(event, context)
+    with caplog.at_level(logging.DEBUG, logger="eoapi-bootstrap"):
+        db_handler.handler(event, context)
+        for record in caplog.records:
+            assert record.message == "OK - Status code: 200"

